base_layer/p2p/src/services/liveness/error.rs

```diff
@@ -42,4 +42,8 @@ pub enum LivenessError {
     TransportChannelError(TransportChannelError),
     /// Ping pong type was invalid or unrecognised
     InvalidPingPongType,
+    /// NodeId was already present in list
+    NodeIdExists,
```

Comment:
I prefer rather than returning an error in this case to simply return a success (and don't add the peer again) - making the call idempotent

Reply:
Alrighty